framework-cloud/framework-openshift/pom.xml

```diff
@@ -22,6 +22,10 @@
       <groupId>org.kie.cloud</groupId>
       <artifactId>framework-cloud-common</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.kie.cloud</groupId>
+      <artifactId>framework-git</artifactId>
```

Comment:
My intention was to keep framework-openshift independent from framework-git, allowing test modules to import just parts of the framework which are needed.
Can you remove the git dependency from here and keep it rather in test module?

Reply:
I provided a solution to remove this dependency using ServiceLoader framework. 

The latest decisions have been made:

- Update GitProviderService to use the service loader too. This service is still in use in the performance tests and test initializer. 
- Move Gogs deployed into the test-remote-cloud and change it to implement the GitProviderFactory instead of being an utility. 
- Since the Gogs provider factory needs the Project to deploy into Openshift, I created a new interface ProjectInitializer to initialize all the services that need this Project instance. This might be used for other service as well.